### ISSUE TITLE
DirectX12 Support

### ISSUE THREAD

**Issue by Mayhem50** [MicrosoftDocs/mixed-reality#406]
When do you release DirectX 12 support for your Mixed Reality ?
I am really pissed off cause I have an UWP App that run a 3D engine on D3D12 but I can't use any HMD due to UWP Sandbox that blocks SteamVR and your SDK that only get support for DirectX11.

**Comment by mattzmsft** [comment]
@thetuvix @MikeRiches any comments?

**Comment by thetuvix** [comment]
Hi @Mayhem50,

There are some key graphics card driver updates that need to get in place before we can add full D3D12 support to Windows Mixed Reality, and we're chasing those down now.  Until then, you can wire up your D3D12 engine by producing your left/right eye textures and then copying them into the D3D11 buffers that Windows Mixed Reality provides.  If your engine has a final MSAA resolve step, you should be able to directly resolve your D3D12 buffer into that target D3D11 buffer without an extra copy.

We'll let you know when there is direct D3D12 platform support for you to try out here!

Alex

**Comment by Mayhem50** [comment]
Hi Alex,

Thanks for the answer. This is what I had supposed. I can do a D3D11On12 to do this copy but it's a non-negligible extra work for me.

When do you think the support of D3D12 will be out ? 1 month ,6 or one year ?

I give you a screenshot on what I am working. 

![image](https://user-images.githubusercontent.com/3661998/40351486-d30e6aba-5dac-11e8-8189-7a9f0a2d4f0c.png)

**Comment by jessemcculloch** [comment]
@thetuvix - Is there any update here?  I'm going through issues trying to close what is no longer relevant.

**Comment by twhall-umich** [comment]
I find that the Oculus OpenXR runtime has better support for D3D12 than the Windows Mixed Reality runtime.  If you have an Oculus Rift and Oculus software, SetEnvironmentVariable ("XR_RUNTIME_JSON", "C:\Program Files\Oculus\Support\oculus-runtime\oculus_openxr_64.json") before calling any OpenXR functions.  The Oculus runtime works flawlessly with our D3D12 renderers, whereas the WMR runtime is prone to crash in a low-level implementation of xrEndFrame:

![xrEndFrame-Crash-in-WMRruntime-from-MobileRenderer](https://user-images.githubusercontent.com/20173113/85346055-43100580-b4c2-11ea-805a-ec55f09d0e83.PNG)

**Comment by hferrone** [comment]
Hi @Mayhem50,

We now have an entire section on DirectX for Mixed Reality at https://review.docs.microsoft.com/en-us/windows/mixed-reality/coordinate-systems-in-directx?branch=pr-en-us-2283. 

Cheers,

**Comment by twhall-umich** [comment]
On Tue, Jun 23, 2020 at 1:57 PM Harrison Ferrone <notifications@github.com>
wrote:

> Hi @Mayhem50 <https://github.com/Mayhem50>,
>
> We now have an entire section on DirectX for Mixed Reality at
> https://review.docs.microsoft.com/en-us/windows/mixed-reality/
> coordinate-systems-in-directx?branch=pr-en-us-2283.
>

Thanks, but I cannot access that page:

   403: Forbidden
In any case, my problem is unrelated to "coordinate-systems-in-directx".  I
think perhaps you have misdirected your message.  (As far as I know, I am
not known as "Mayhem50")

-- 

Ted Hall <twhall@umich.edu>
Virtual Reality Visualization Specialist
Duderstadt Center
University of Michigan
(734) 647-5745
*https://www.dc.umich.edu/ <https://www.dc.umich.edu/>*

**Comment by hferrone** [comment]
Hi @twhall-umich, 

I was responding to the original poster with the previous message. Thanks for your message.

Cheers,
